packages/server/src/manifest-builder.ts

```diff
@@ -24,14 +24,33 @@ interface ManifestBuilderOptions {
   distDir: string;
 };
 
+function ftruncate(fd: number, len: number): Operation {
+  return ({ fail, resume }) => {
+    fs.ftruncate(fd, len, err => {
+      if (err) {
+        fail(err);
+        return;
+      }
+      resume();
+    })
+  };
+}
```

Comment:
We want to avoid using the "control function" version of operations since they will almost certainly be deprecated in the next version of effection. Instead, we can use a `Deferred` object to convert any callback-based api into a promise which can then be used as a normal operation.
```ts
import { Deferred } from '@bigtest/effection';

function*  ftruncate(fd: number, len: number): Operation<void> {
  let { resolve, reject, promise } = Deferred<void>();
  fs.ftruncate(fd, len, err => {
    if (err) {
      reject(err);
    } else {
      resolve();
    }
  });
  yield promise;
}
```